content/posts/2021-01-08-gradle-custom-shortcut.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "Stop wasting your Android gradle daemons"
+date: 2021-01-08T16:48:02+09:00
+draft: false
+---
+
+As you probably noticed, switch between your dear Android Studio and your terminal
+is not without some latency / weird error messages about some Gradle daemons that could not be reused.
+
+{{< image src="/assets/2021-01-08-gradle-custom-shortcut/not-reused.png" position="center" >}}
+
+The reason is because Android Studio is coming by default with its own JDK installation,
+allowing to use Android Studio without having to install Java
+or to be sure all your team is using the same compiler.
+
+That's convenient for sure, but it's not something your Gradle project is aware and so,
+if you're trying to use Gradle without Android Studio's help,
+you will probably fallback to your default system JDK,
+which is not the one that Android Studio is using.
+It can affects you in 2 ways:
+
+1. not re-using the existing Gradle daemon (!!)
+1. not re-using your build cache (if your default JDK has a different version than the AS's one)
```

Comment:
Not an English comment but this is the cause rather than the effect 😅Instead I'd like to know what is the effect of not re-using the existing gradle daemon. Does it make things slower? Use extra resources? Compile differently? The current statement doesn't really mean anything to me (or other people not familiar with the gradle process).

Reply:
I guess I should replace this fake list by a block just explaining that the Gradle daemon is good for you, and redirect to https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:why_the_daemon to know more

Reply:
@EmmaKiwi please take a look at https://github.com/pgreze/blog/pull/16/files#r554508357 🙏